pytorch_lightning/trainer/trainer.py

```diff
@@ -284,10 +285,15 @@ def __init__(
 
             benchmark: If true enables cudnn.benchmark.
 
+            deterministic: If true enables cudnn.deterministic
+
             terminate_on_nan: If set to True, will terminate training (by raising a `ValueError`) at the
                 end of each training batch, if any of the parameters or the loss are NaN or +/-inf.
         """
 
+        self.deterministic = deterministic
+        torch.backends.cudnn.deterministic = self.deterministic
```

Comment:
does it mean that TPU or Horovod doesn't have deterministic?
@tgaddair ^^

Reply:
We do not have a single flag to enable determinism, that's correct.  The only non-deterministic part of Horovod is tensor fusion, which can be disabled on the command-line, or by setting `os.environ['HOROVOD_FUSION_THRESHOLD'] = '0'` before calling `hvd.init()`.

Reply:
@kumuji an you add it there..

Reply:
Added to the Trainer class where deterministic sets.